splice/web/api/campaign.py

```diff
@@ -126,5 +131,40 @@ def put(self, campaign_id):
 api.add_resource(CampaignAPI, '/campaigns/<int:campaign_id>', endpoint='campaign')
 
 
+_VALID_CREATIVE_EXTENSIONS = set(['zip'])
+_VALID_ASSETS_EXTENSIONS = set(['tsv', 'txt'])
+
+
+@campaign_bp.route('/campaigns/<int:campaign_id>/bulkupload', methods=['POST'])
+def handler_bulk_upload(campaign_id):
+    """bulk uploading adgroups and tiles with the given campaign_id."""
+
+    def is_allowed_files(name, allowed_sets):
+        return '.' in name and name.rsplit('.', 1)[1] in allowed_sets
+
+    creatives = request.files["creatives"]
+    assets = request.files["assets"]
+    if not is_allowed_files(creatives.filename.lower(), _VALID_CREATIVE_EXTENSIONS) \
+            or not is_allowed_files(assets.filename.lower(), _VALID_ASSETS_EXTENSIONS):
+            return jsonify(message="Invalid files uploaded."), 400
+
+    campaign = get_campaign(campaign_id)
+    if campaign is None:
+        return jsonify(message='Campaign not found.'), 404
+
+    (_, tmp_creatives) = tempfile.mkstemp()
+    (_, tmp_assets) = tempfile.mkstemp()
+    creatives.save(tmp_creatives)
+    assets.save(tmp_assets)
```

Comment:
Actually, you are saving these files to disk and reopening them again down in load assets. Why not just pass the files down. `zipfile` can take a file object instead of a path. And you can avoid doing the `open(assets, "rb") as f`. Would probably make everything a bit faster.